servers/rendering/rendering_device.cpp

```diff
@@ -170,10 +170,16 @@ RenderingDevice::VertexFormatID RenderingDevice::_vertex_format_create(const Typ
 	return vertex_format_create(descriptions);
 }
 
-RID RenderingDevice::_vertex_array_create(uint32_t p_vertex_count, VertexFormatID p_vertex_format, const TypedArray<RID> &p_src_buffers) {
+RID RenderingDevice::_vertex_array_create(uint32_t p_vertex_count, VertexFormatID p_vertex_format, const TypedArray<RID> &p_src_buffers, const Vector<int64_t> &p_offsets) {
 	Vector<RID> buffers = Variant(p_src_buffers);
 
-	return vertex_array_create(p_vertex_count, p_vertex_format, buffers);
+	Vector<uint64_t> offsets;
+	offsets.resize(p_offsets.size());
+	for (int i = 0; i < p_offsets.size(); i++) {
+		offsets.write[i] = p_offsets[i];
+	}
```

Comment:
I wonder if there is a trivial conversion between ``Vector<int64_t>`` and ``Vector<uint64_t>``. Perhaps we could even cast here to avoid doing a full copy.

Reply:
There's a potential horrible hack possible. With some static assertion to ensure both have the same layout, just `reinterpret_cast` one into the other.

Reply:
I couldn't find a simpler/better way to do the conversion, but I'm certainly open to suggestions. An ugly hack doesn't really seem worth it for a scripting interface that's dealing with fairly small arrays.

Reply:
I guess the length of ``offsets`` is never greater than ``ARRAY_MAX``, right? Which means it will never be very long. So probably not a big deal

Reply:
The hard limit is `VkPhysicalDeviceLimits::maxVertexInputBindings`, which seems to be 16 - 32.

Reply:
Okay, and in internal code, the largest we ever use is 12 (in the case of a 3D mesh with all attributes). Its safe to say the loop will always be short and we are only copying at most 32 values. This is a fine approach.